akka-http-core/src/main/scala/akka/http/scaladsl/model/HttpMessage.scala

```diff
@@ -160,6 +168,12 @@ sealed trait HttpMessage extends jm.HttpMessage {
       case _                 ⇒ Optional.empty()
     }
   /** Java API */
+  def getHeaders[T <: jm.HttpHeader](headerClass: Class[T]): JIterable[T] = {
+    headers.collect {
+      case h if headerClass.isInstance(h) ⇒ h.asInstanceOf[T]
+    }.asJava
+  }
```

Comment:
As I mentioned, I'm not really happy to duplicate this bit. But I can't make compiler happy if I just call `headers[T]` as the type T is slightly different on the Scala counterpart.

Reply:
With the `>: Null` type bound gone, this can be:

```scala
  /** Java API */
  def getHeaders[T <: jm.HttpHeader](headerClass: Class[T]): JIterable[T] =
    headers[T](ClassTag[T](headerClass)).asJava
```